org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/EclipseResourceFileSystemAccess2.java

```diff
@@ -533,12 +533,15 @@ protected IFile getTraceFile(IFile file) {
 	}
 
 	private void syncIfNecessary(IResource result) {
-		syncIfNecessary(result, monitor);
+		syncIfNecessary(result, this.monitor);
 	}
 
 	private void syncIfNecessary(IResource result, IProgressMonitor progressMonitor) {
 		try {
-			sync(result, IResource.DEPTH_ZERO, progressMonitor);
+			// org.eclipse.core.internal.resources.Resource.refreshLocal(int, IProgressMonitor) starts a new subtask
+			// that we don't want to propagate to the parent monitor.
+			SubMonitor subMonitor = SubMonitor.convert(monitor, 1);
+			sync(result, IResource.DEPTH_ZERO, subMonitor.split(1, SubMonitor.SUPPRESS_ALL_LABELS));
```

Comment:
Should the call of syncIfNecessary already suppress the labels?

Reply:
sync comes from the ResourceUtil that we don't want to change -> so no other place to do that and sync always calls Resource.refreshLocal.